# Contributor: Connor Behan <connor.behan@gmail.com>

pkgname=pida-hg
pkgver=2920
pkgrel=1
pkgdesc="Python Integrated Development Application"
arch=('i686' 'x86_64')
url="http://pida.co.uk/"
license=('custom')
depends=('vte' 'python-simplejson' 'pygtkhelpers-hg' 'dbus-python' 'python2-argparse' 'python2-logbook' 'python-notify')
makedepends=('mercurial' 'python2-distribute')
optdepends=('anyvc: version control integration')
provides=('pida')
conflicts=('pida')
source=(pida.desktop nodrm.patch internal_moo.patch api_update.patch)
install=pida.install
# Do you get 404 errors if you use the standard "_hgroot" and "_hgrepo"?
__hgroot=http://bitbucket.org/aafshar/pida-main
__hgrepo=pida

build() {
  cd $srcdir
  
  msg "Connecting to $__hgroot"
  if [ -d $__hgrepo ]; then
    cd $__hgrepo
    hg pull -u || return 1
    msg2 "Finished updating the local repository!"
  else
    hg clone $__hgroot $__hgrepo || return 1
    msg2 "Initial pull complete!"
  fi

  rm -rf ${srcdir}/${__hgrepo}-build
  cp -a ${srcdir}/${__hgrepo} ${srcdir}/${__hgrepo}-build || return 1
  cd ${srcdir}/${__hgrepo}-build
  
  patch -Np1 -i $srcdir/nodrm.patch || return 1
  patch -Np1 -i $srcdir/internal_moo.patch || return 1
  patch -Np1 -i $srcdir/api_update.patch || return 1
  python2 setup.py install --root=$startdir/pkg/ --prefix=/usr

  install -Dm644 COPYING \
    $startdir/pkg/usr/share/licenses/pida/LICENSE
  
  install -Dm644 pida/resources/pixmaps/pida-icon.png \
    $startdir/pkg/usr/share/pixmaps/pida.png
  install -Dm644 $startdir/src/pida.desktop \
    $startdir/pkg/usr/share/applications/pida.desktop
}
md5sums=('0216466cf5f64238c90b5e07df71ce3a' '479c24a04069112f610667f9746011d4' '8dd22f63476c8f5ae7736e2e67bb50d1' '5f9d5dbbd6f008278847122dd368d8c4')
